_posts/2018-04-15-lazy-loading-images.md

```diff
@@ -0,0 +1,205 @@
+      <img class="image__list-image image__list-lazy-image" data-src="http://res.cloudinary.com/dsteinel/image/upload/c_scale,w_450/v1523796305/pexels-photo-104336.jpg"
+        alt="street in the middle of a forrest">
+    </li>
+    <li class="image__list-item">
+      <img class="image__list-image image__list-lazy-image" data-src="http://res.cloudinary.com/dsteinel/image/upload/c_scale,w_450/v1523796376/les-anderson-175603.jpg"
+        alt="two elder people opening togehter a christmas present">
+    </li>
+    <li class="image__list-item">
+      <img class="image__list-image image__list-lazy-image" data-src="http://res.cloudinary.com/dsteinel/image/upload/c_scale,w_450/v1523796381/pexels-photo-737108.jpg"
+        alt="black and white clothes hanging on a clothes rail">
+    </li>
+    <li class="image__list-item">
+      <img class="image__list-image image__list-lazy-image" data-src="http://res.cloudinary.com/dsteinel/image/upload/c_scale,w_450/v1523796475/tobias-van-schneider-122288.jpg"
+        alt="book collection photographed from top">
+    </li>
+    <li class="image__list-item">
+      <img class="image__list-image image__list-lazy-image" data-src="http://res.cloudinary.com/dsteinel/image/upload/c_scale,w_450/v1523796300/pexels-photo-559768.jpg"
+        alt="brown meadow with a tower in the very back">
+    </li>
+  </ul>
+</div>
+{% endhighlight %}
+
+{% highlight css %}
+.image-container {
+  margin: 0 auto;
+  width: 450px;
+}
+
+.image__list-lazy-image { 
+  display: none;
+}
+
+.image__list-image {
+  display: block;
+}
+
+.image__list-item {
+  margin-bottom: 40px;
+}
+{% endhighlight %}
+
+Let’s get to JavaScript part. There is a very nice little helper which was announced on the 14 September’17 by Google. They describe it as follows:
+"This specification describes an API that can be used to understand the visibility and position of DOM elements ("targets") relative to a containing element or to the top-level viewport ("root"). The position is delivered asynchronously and is useful for understanding the visibility of elements and implementing pre-loading and deferred loading of DOM content.” [source](https://www.w3.org/TR/intersection-observer/)
+
+**This is just perfect for lazy loading images!**
+
+So we first have to gather all the images we marked above as “lazy-images”. Those are the images which have a data attribute which carries our image source. Make sure that the image has no src attribute set, otherwise, the browser will load an image before the IntersectionObserver can do its magic. The logic result we are now trying to achieve is that no lazy load image has an image source, meaning that no image will be loaded initially. The imageObserver will now check on each image, if it is in the viewport. If it is, then we will replace the content of the src attribute with the content of the data-src attribute (the actual image source). If this is done, we will download and display the image.
```

Comment:
Those are the images that have a data attribute containing our image source

Reply:
Use markdown code highlighting:

Make sure that the image has no `src` attribute set